### ISSUE TITLE
Add a proper definition of additional_hash

### ISSUE THREAD

**Issue by loredanacirstea** [raiden-network/spec#62]
Add a proper definition here:
https://github.com/raiden-network/spec/blob/master/messaging.rst#balance-proof

`Computed from the message. Used for message authentication` is superficial.

Definition should make the following clear:

- when and where is this used
- who calculates it and from what data (mention if this is an unknown or can be customized; give an example)
- who must verify this
- if any 3rd party services must/could use it in some way

Definition details from RC discussions & other docs:

- Because it's not necessarily the hash of the message, it may be zeroed out or just the hash of a subset of the fields. This field is used to add support for authentication without required multiple layers of signatures. Think of the metadata used for the payment, the metadata is unspecified in this spec and the application may want to authenticated it nevertheless.

- Adding salt data to the messages, the goal is to always randomly change `additional_hash` and make it harder for a monitoring service to brute force the hashed balance proof

**Comment by hackaugusto** [comment]
`additional_hash` was intentionally vaugly defined, it's used to authenticate additional data *not* speced. Example: the payment metadata that is defined by the user application.

**Comment by LefterisJP** [comment]
@hackaugusto So in short it will be a hash of some other data, e.g. payment metadata used by the application and its purpose would be to verify said metadata?

**Comment by hackaugusto** [comment]
yep

**Comment by LefterisJP** [comment]
Yeah so that is what I explained to @palango the last time we talked about it. I think this is a sufficient explanation, left intentionally vague since it's something to facilitate the apps built on top of raiden, and not the core protocol itself.

@loredanacirstea I opened this: https://github.com/raiden-network/spec/pull/69

Now not all of your questions are answered there and example is not given, at least not in detail but the suggested place is a table where a short definition should be given. Nothing more elaborate.

If you want more elaborate definition I can add it to the terminology.rst. Review the PR and give me your opinion. 

@palango I would also appreciate your opinion on this.
